runtime/onert/core/src/util/shapeinf/StridedSlice.cc

```diff
@@ -0,0 +1,301 @@
+{
+  ir::Shape out_shape;
+  int32_t shape_size = 0;
+
+  for (uint32_t idx = 0; idx < rank; ++idx)
+  {
+    int32_t stride = op_params.strides[idx];
+    int32_t begin = StartForAxis(op_params, input_shape, idx);
+    int32_t end = StopForAxis(op_params, input_shape, idx, begin);
+
+    // When shrinking an axis, the end position does not matter (and can be
+    // incorrect when negative indexing is used, see Issue #19260). Always use
+    // begin + 1 to generate a length 1 slice, since begin has
+    // already been adjusted for negative indices by StartForAxis.
+    const bool shrink_axis = op_params.shrink_axis_mask & (1 << idx);
+    if (shrink_axis)
+    {
+      end = begin + 1;
+    }
+
+    int32_t dim_shape = std::ceil((end - begin) / static_cast<float>(stride));
+    dim_shape = dim_shape < 0 ? 0 : dim_shape;
+    if (!shrink_axis)
+    {
+      shape_size++;
+      out_shape.append(dim_shape);
+    }
+  }
+
+  return out_shape;
+}
+
+void StaticInferer::visit(const ir::operation::StridedSlice &op)
+{
+
+  const auto input_index{op.getInputs().at(ir::operation::StridedSlice::Input::INPUT)};
+  const auto &input = _operands.at(input_index);
+  const auto starts_index{op.getInputs().at(ir::operation::StridedSlice::Input::STARTS)};
+  const auto &starts = _operands.at(starts_index);
+  const auto ends_index{op.getInputs().at(ir::operation::StridedSlice::Input::ENDS)};
+  const auto &ends = _operands.at(ends_index);
+  const auto strides_index{op.getInputs().at(ir::operation::StridedSlice::Input::STRIDES)};
+  const auto &strides = _operands.at(strides_index);
+  const auto output_index = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_index);
+
+  if (input.info().isDynamic())
+  {
+    output.info().setDynamic();
+    return;
+  }
+
+  if (!(starts.isConstant() && ends.isConstant() && strides.isConstant()))
+  {
+    output.info().setDynamic();
+    return;
+  }
+
+  const auto begin_mask = op.param().begin_mask;
+  const auto end_mask = op.param().end_mask;
+  const auto shrink_axis_mask = op.param().shrink_axis_mask;
+  const auto rank = op.param().rank;
+
+  auto starts_buf = reinterpret_cast<const uint32_t *>(starts.data()->base());
+  auto ends_buf = reinterpret_cast<const uint32_t *>(ends.data()->base());
+  auto strides_buf = reinterpret_cast<const uint32_t *>(strides.data()->base());
+
+  auto op_params = buildStridedSliceParams(starts_buf, ends_buf, strides_buf, begin_mask, end_mask,
+                                           shrink_axis_mask, rank);
+
+  ir::Shape new_shape = inferStridedSliceShape(input.info().shape(), op_params, rank);
+  output.info().shape(new_shape);
+}
+
+void DynamicInferer::visit(const ir::operation::StridedSlice &op)
+{
+
+  const auto input_index{op.getInputs().at(ir::operation::StridedSlice::Input::INPUT)};
+  auto input = _tensor_registry->getITensor(input_index);
+  ir::Shape input_shape = getShape(input.get());
+
+  if (!input->is_dynamic())
```

Comment:
If `input` was static but, e.g., `ends` was an output of an op, `output` is dynamic (set by static tensor inferer).
However in this method, dynamic shape inferer does simply returns without allocating the output memory.